### DISCUSSION TITLE
Set workspace to specific monitor

### DISCUSSION THREAD

**Discussion by winterec** [regolith-linux/regolith-desktop#939] [Q&A]
I'm having trouble assigning a workspace to a specific output. I'm seeing the same error regardless of quotes used.

Here is my i3 config
`workspace "$ws3" output DP-4`

Here is the error I'm seeing

```ERROR: CONFIG: Expected one of these tokens: 'output', 'gaps'
ERROR: CONFIG: (in file /home/edward/.config/regolith3/i3/config.d/01-autostart)
ERROR: CONFIG: Line   1: workspace "3:<span> </span>3 <span foreground="#859900"></span><span> </span>" output DP-4
ERROR: CONFIG:                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: CONFIG: Line   2: 
ERROR: CONFIG: Line   3: assign [class="^obsidian$"] 3:<span> </span>3 <span foreground="#859900"></span><span> </span>
ERROR: FYI: You are using i3 version 4.21 (2022-09-21)
ERROR: FYI: You are using i3 version 4.21 (2022-09-21)
ERROR: i3-nagbar (156436) did not exit normally. This is not an error if the config was reloaded while a nagbar was active.
```

I have the default workspace name with the coloured circle label. Here is the output from regolith-diagnostic

```
wm.workspace.03.name:	3:<span> </span>3 <span foreground="#859900"></span><span> </span>
```

I've been stuck on this for a little while and would appreciate any help

**Comment by limefrog-the-genuine-one** [comment]
Hi @winterec,

i just want to ask,if you found a solution. Searching for a way to assign a workspace to a specific physical screen and assigning an application to a workspace either (thus assigning applications to physical screens), i stumbled across your post.

Regards

**Comment by pmahnke** [comment]
Are you running sway or x?

**Comment by limefrog-the-genuine-one** [comment]
Hi,

I'm running X on debian 12 and got the assignment of screens and workspaces running with:

```
workspace "$ws1" output DP-2.1.1.8
workspace "$ws2" output DP-3.8
workspace "$ws3" output DP-2.1.8
workspace "$ws4" output DP-2.1.1.8
workspace "$ws5" output DP-3.8
workspace "$ws6" output DP-2.1.8
workspace "$ws7" output DP-2.1.1.8
workspace "$ws8" output DP-3.8
workspace "$ws9" output DP-2.1.8
workspace "$ws10" output DP-2.1.8
workspace "$ws19" output DP-2.8
```

but i didn't manage to assign applications to workspaces:

```
assign [title="thunderbird"] "$ws1"
assign [title="firefox-esr"] "$ws2"
```

This didn't work...

... and very strange: mod+shif+c resets my background from 'zoom' to 'wallpaper' and changes my iconset and theme also!

**Comment by pmahnke** [comment]
Good news on the workspaces.

As for the assign, I have had more luck using `class` instead... like:

`assign [class="Firefox"] "$ws2"`

But I am no expert, I don't know why.

**Comment by limefrog-the-genuine-one** [comment]
Yep!

That does the trick, but i really want to know WHY?

;-)

**Comment by winterec** [comment]
Hi friends,

@limefrog-the-genuine-one Your workspace lines look similar to mine but with a the decimals for the workspace numbers. When I check xrandr now the screen I want to use is actually given the identifier DP-1-4. I tried that and DP-1.4 and it still didn't seem to work. 

I'm still getting the same error which seems to be related to the html tags in the string that $ws3 expands to. Did you ever have something similar to this? I think I'm using the base regolith theme which has coloured circles in the workspace tabs.

@pmahnke I'm using X on Ubuntu with LightDM as well.

**Comment by limefrog-the-genuine-one** [comment]
@winterec: For me it still works but nevertheless if I restart the machine I get an error message, which disappears once I restart the environment. Since the machine runs about two or three months that doesn't bother me too much.

**Comment by notmii** [comment]
A work around for me is to remove the icons

Added the following in `$HOME/.config/regolith3/Xresources`
```
wm.workspace.01.name: 1
wm.workspace.02.name: 2
wm.workspace.03.name: 3
wm.workspace.04.name: 4
wm.workspace.05.name: 5
wm.workspace.06.name: 6
wm.workspace.07.name: 7
wm.workspace.08.name: 8
wm.workspace.09.name: 9
```

Added the following in `$HOME/.config/regolith3/i3/conf.d/90_config`
```
workspace "$ws1" output DVI-I-3-2
workspace "$ws2" output DVI-I-3-2
workspace "$ws3" output DVI-I-3-2
workspace "$ws4" output DVI-I-3-2
workspace "$ws5" output DVI-I-2-1
workspace "$ws6" output DVI-I-2-1
```

**Comment by dariooddenino** [comment]
I'm having the same issue.
I get that error, which goes away after restarting regolith with mod+shift+r.
After that I have to re-select workspaces to create the correct one with icons.

I'm on ubuntu 24.04, regolith 3.2 and i3 4.23

My config is very minimal:
```
# Assign workspaces to monitors
workspace "$ws1" output DP-1
workspace "$ws2" output DP-1
workspace "$ws3" output DP-1
workspace "$ws4" output DP-1
workspace "$ws5" output DP-1
workspace "$ws6" output DP-2
workspace "$ws7" output DP-2
workspace "$ws8" output DP-2
workspace "$ws9" output DP-2
workspace "$ws10" output DP-2
workspace "$ws11" output HDMI-0
workspace "$ws12" output HDMI-0
workspace "$ws13" output HDMI-0
workspace "$ws14" output HDMI-0
workspace "$ws15" output HDMI-0

# Assign clients to workspaces
# First monitor
assign [class="^firefox$"] "$ws1"
assign [class="^Spotify$"] "$ws5"
# Third monitor
assign [class="^Slack$"] "$ws11"
assign [class="^discord$"] "$ws11"
```

**Reply by ingsme** [reply]
I think you are missing 'workspace'. Like this: assign [whatever] **workspace** "$ws1"
This works for me:

```
# Set workspace and monitor
set $ws1 1
set $ws2 2
set $ws3 3
set $monitor1 DP-1-8
set $monitor2 DP-2
set $monitor3 HDMI-1

# Assign workspace to monitor
workspace "$ws1" output "$monitor1"
workspace "$ws2" output "$monitor2"
workspace "$ws3" output "$monitor3"

# Place apps on workspaces
assign [class="(?i)firefox"] workspace "$ws3"
assign [class="(?i)microsoft (?i)teams"] workspace "$ws2"
```

**Reply by dariooddenino** [reply]
I'm not using the "assign clients to workspace" part. It probably doesn't work, but the problem is with assigning the workspaces to the monitors, which only works after I restart regolith after a reboot.

**Comment by idlethread** [comment]
I see similar issues with the following errors:

```
ERROR: CONFIG: Expected one of these tokens: 'output', 'gaps'
ERROR: CONFIG: (in file /home/idlethread/.config/regolith3/i3/config.d/custom)
ERROR: CONFIG: Line  18: #set 10:<span> </span>10 <span foreground="#268bd2"></span><span> </span> 10
ERROR: CONFIG: Line  19: 
ERROR: CONFIG: Line  20: workspace "1:<span> </span>1 <span foreground="#268bd2"></span><span> </span>" output "HDMI-A-0"
ERROR: CONFIG:                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: CONFIG: Line  21: #workspace "2:<span> </span>2 <span foreground="#2aa198"></span><span> </span>" output "HDMI-A-0"
ERROR: CONFIG: Line  22: #workspace "3:<span> </span>3 <span foreground="#859900"></span><span> </span>" output "HDMI-A-0"
ERROR: FYI: You are using i3 version 4.23 (2023-10-29)

```

Replacing the definition of ws1 with a simpler 

`set $ws1 1
`

seem to make the issues go away. It seems like the config parser doesn't seem to like the extra decoration for some reason. This used to work earlier on Ubuntu 22.04 packages.
